### DISCUSSION TITLE
Admin 2FA and AppAmour questions

### DISCUSSION THREAD

**Discussion by Stephan-4711** [StrausFuenf/hassio-addons#8] [General]
Hi,

what is the new 2FA for the Admin panel and why only for new installations?
and how to configure Appamour?

**Comment by strausfuenf** [comment]
Hey,

its not 2FA in the normal community addon the Admin Token is generated as a plain text token which brings up this message:

[NOTICE] You are using a plain text `ADMIN_TOKEN` which is insecure.
Please generate a secure Argon2 PHC string by using `vaultwarden hash` or `argon2`.

I changed the initial setup to generate a secure Admin Token which only works if there is no config.json.

To fix it in a running config you need to generate one yourself (https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token)

Apparmor needs no configuration its a security feature to even more secure the container so no harmful code can be executed on the host. (https://www.apparmor.net/)
